name.abuchen.portfolio/src/name/abuchen/portfolio/model/ClientSettings.java

```diff
@@ -0,0 +1,142 @@
+package name.abuchen.portfolio.model;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import name.abuchen.portfolio.online.WebLocation;
+
+public class ClientSettings implements PropertyChangeListener
+{
+    private static Map<Client,ClientSettings> instances = new HashMap<Client,ClientSettings>();
+    
+    private Client client;
+    private ArrayList<WebLocation> bookmarksData;
+
+    // Return the ClientSettings Instance for given client
+    public static ClientSettings getInstance(Client client){
+        ClientSettings c = instances.get(client);
+        
+        if ( c == null ) {
+            c = new ClientSettings(client);
+            instances.put(client, c);
+        }
+        
+        return c;
+    }
+    
+    public ClientSettings(Client client)
+    {
+        this.client = client;
+        this.client.addPropertyChangeListener(this);
+        this.bookmarksData = new ArrayList<WebLocation>();
+        loadBookmarks();        
+    }
+    
+    public void setDefaultBookmarks() {
+        
+        bookmarksData.add(new WebLocation("OnVista", "http://www.onvista.de/suche.html?SEARCH_VALUE={isin}&SELECTED_TOOL=ALL_TOOLS")); //$NON-NLS-1$
+        bookmarksData.add(new WebLocation("Finanzen.net", //$NON-NLS-1$
+                        "http://www.finanzen.net/suchergebnis.asp?frmAktiensucheTextfeld={isin}")); //$NON-NLS-1$
+        bookmarksData.add(new WebLocation("Ariva.de Fundamentaldaten", //$NON-NLS-1$
+                        "http://www.ariva.de/{isin}/bilanz-guv")); //$NON-NLS-1$
+        bookmarksData.add(new WebLocation("justETF", //$NON-NLS-1$
+                        "https://www.justetf.com/de/etf-profile.html?isin={isin}")); //$NON-NLS-1$
+        bookmarksData.add(new WebLocation("fondsweb.de", //$NON-NLS-1$
+                        "http://www.fondsweb.de/{isin}")); //$NON-NLS-1$
+        bookmarksData.add(new WebLocation("Morningstar.de", //$NON-NLS-1$
+                        "http://www.morningstar.de/de/funds/SecuritySearchResults.aspx?type=ALL&search={isin}")); //$NON-NLS-1$
+        bookmarksData.add(new WebLocation(
+                        "maxblue Kauforder", //$NON-NLS-1$
+                        "https://meine.deutsche-bank.de/trxm/db/init.do" //$NON-NLS-1$
+                                        + "?style=mb&style=mb&login=br24order&action=PurchaseSecurity2And3Steps&wknOrIsin={isin}")); //$NON-NLS-1$         
+    }
+    public void loadBookmarks() {
```

Comment:
Bei mir hat das Laden der Bookmarks nicht funktioniert. Im Debugger habe ich dann gesehen, dass ich folgende Exception bekomme:
```
java.io.StreamCorruptedException: invalid stream header: EFBFBDEF
```

Zunächst wäre es super einfach mit ```PortfolioPlugin.log(e)``` die Exceptions wegzuschreiben. Der Benutzer sieht das ja nicht, aber wenn nachfragen kommen, hat man eine Chance zu verstehen, was abgeht.

Dann würde ich vorschlagen *keine* serialisierten Objekte im XML zu speichern. Erstens binden wir uns dann an die Sprache Java (was ist mit der zukünftigen iOS Version? :wink:). Zweitens läuft man leicht in solche Serialisierungsprobleme rein. Und dann finde ich es hilfreich zur Not im Texteditor Dinge korrigieren zu können.

Die loadBookmarks und saveBookmarks Methoden könnten aber einfach einen String generieren. Man muss etwas auf das Esapting achten.